### PR TITLE
Make all our dataclasses frozen

### PR DESCRIPTION
**Pull request by PerchunPak**
This fixes some of the errors that pyright [outputs](https://github.com/py-mine/mcstatus/actions/runs/7789626721/job/21241666878?pr=741) after an update:
```
/home/runner/work/mcstatus/mcstatus/mcstatus/status_response.py
  /home/runner/work/mcstatus/mcstatus/mcstatus/status_response.py:109:5 - error: "players" overrides symbol of same name in class "BaseStatusResponse"
    Variable is mutable so its type is invariant
      Override type "JavaStatusPlayers" is not the same as base type "BaseStatusPlayers" (reportIncompatibleVariableOverride)
  /home/runner/work/mcstatus/mcstatus/mcstatus/status_response.py:110:5 - error: "version" overrides symbol of same name in class "BaseStatusResponse"
    Variable is mutable so its type is invariant
      Override type "JavaStatusVersion" is not the same as base type "BaseStatusVersion" (reportIncompatibleVariableOverride)
  /home/runner/work/mcstatus/mcstatus/mcstatus/status_response.py:153:5 - error: "players" overrides symbol of same name in class "BaseStatusResponse"
    Variable is mutable so its type is invariant
      Override type "BedrockStatusPlayers" is not the same as base type "BaseStatusPlayers" (reportIncompatibleVariableOverride)
  /home/runner/work/mcstatus/mcstatus/mcstatus/status_response.py:154:5 - error: "version" overrides symbol of same name in class "BaseStatusResponse"
    Variable is mutable so its type is invariant
      Override type "BedrockStatusVersion" is not the same as base type "BaseStatusVersion" (reportIncompatibleVariableOverride)
```

This is probably a breaking change (you can't change the values of returned objects anymore), although we already have a breaking change (#682) in our master, so it is not a big deal I think.

### CODE REVIEW

File: tests/status_response/test_java.py [hunk 1]
```diff
@@ -88,11 +88,10 @@ def build(self):
...
 
     def test_id_field_the_same_as_uuid(self):
-        build = JavaStatusPlayer.build({"name": "foo", "id": "0b3717c4-f45d-47c8-b8e2-3d9ff6f93a89"})
+        unique = object()
+        build = JavaStatusPlayer.build({"name": "foo", "id": unique})  # type: ignore[assignment]
         assert build.id is build.uuid
-
-        build.id = unique = object()  # type: ignore[assignment]
-        assert unique is build.uuid
+        assert build.uuid is unique
```
Comment (ItsDrike) [inline tests/status_response/test_java.py:94]: How is this related to the PR's topic?

**Review by ItsDrike — CHANGES_REQUESTED** [review]
Yes, we do have a breaking change in master, however it's just a deprecation removal, the change itself was made a while ago and went through a full deprecation period by now. 

This, in it's current state, would be a breaking change without any deprecation. However, there's no easy way to add deprecations here (it wouldn't be impossible, there's descriptors, but it would be a huge and complex mess, probably with a lot of type-ignores),

That said, I can't imagine this affecting too many people, as there shouldn't really be any need to modify the data from the responses. Because of that, I don't mind it hugely, I just wanted to mention this here, and it should also be made very clear in the release notes.

File: tests/status_response/test_java.py (hunk 1 above)
Comment (PerchunPak) [reply tests/status_response/test_java.py:94]: This test modified dataclass `JavaStatusPlayer`, so after changes it failed.

Comment (ItsDrike) [reply tests/status_response/test_java.py:94]: Oh right, I totally missed that

**Review by ItsDrike — APPROVED** [review]
Assuming we're fine with the breaking change here, this can be merged.
